### PR TITLE
HL-891: Fix CSRF token issues

### PR DESCRIPTION
**Pull request by mjturt**
## Description :sparkles:

Using `/me` endpoint to deliver the CSRF token and saving it to browsers local storage.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2182.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2182.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2182.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2182.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2182.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/applicant/src/hooks/useUserQuery.ts [hunk 1]
```diff
@@ -51,12 +51,14 @@ const useUserQuery = (
       select: (data) => camelcaseKeys(data, { deep: true }),
       onError: (error) => handleError(error),
       onSuccess: (data) => {
+        /* eslint-disable scanjs-rules/identifier_localStorage */
+        localStorage.setItem(LOCAL_STORAGE_KEYS.CSRF_TOKEN, data.csrfToken as string);
```
Comment (sirtawast) [inline frontend/benefit/applicant/src/hooks/useUserQuery.ts:55]: Throws linter error. Maybe just `data.csrfToken` or `String(data.csrfToken)`?

File: frontend/benefit/handler/src/hooks/useUserQuery.ts [hunk 2]
```diff
@@ -46,6 +46,12 @@ const useUserQuery = <T extends User>(
     }
   };
 
+  const onSuccessHandler = (user: User): void => {
+    checkForStaffStatus(user);
+    // eslint-disable-next-line scanjs-rules/identifier_localStorage
+    localStorage.setItem(LOCAL_STORAGE_KEYS.CSRF_TOKEN, user.csrf_token);
```
Comment (sirtawast) [inline frontend/benefit/handler/src/hooks/useUserQuery.ts:52]: These `// eslint-disable-next-line scanjs-rules/identifier_localStorage` start to get out of hand. We should make a `localStorage` **util** or maybe [useLocalStorage](https://www.npmjs.com/package/use-local-storage) sort of thing (don't know if this is THE lib to use).

Comment (sirtawast) [reply frontend/benefit/handler/src/hooks/useUserQuery.ts:52]: I'll make a new ticket out of this.

**Review by sirtawast — CHANGES_REQUESTED** [review]

File: frontend/benefit/handler/src/pages/_app.tsx [hunk 3]
```diff
@@ -33,6 +33,7 @@ const App: React.FC<AppProps> = (appProps) => {
     <BackendAPIProvider
       baseURL={getBackendDomain()}
       headers={getHeaders(locale)}
+      useLocalStorageCsrf
```
Comment (sirtawast) [inline frontend/benefit/handler/src/pages/_app.tsx:36]: This think should be named `isLocalStorageCsrf` as `use` is reserved for hooks.

**Review by sirtawast — CHANGES_REQUESTED** [review]

**Comment by sirtawast** [comment]
Looks good! Fix these small things & merge 🙏

File: frontend/benefit/handler/src/hooks/useUserQuery.ts (hunk 2 above)
Comment (mjturt) [reply frontend/benefit/handler/src/hooks/useUserQuery.ts:52]: Yeah true, I will do this too here

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2182.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2182.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2182.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2182.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2182.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2182.dev.hel.ninja :laughing::tada::tada::tada:

**Review by sirtawast — APPROVED** [review]

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2182.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2182.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2182.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2182.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2182.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**